### PR TITLE
LPS-157175 | Automate LPS-148397 

### PR DESCRIPTION
**Pull request by jaimelr10**
Description: Creation of Priority 3 test to verify that when we download a zip file with a given structure, and we unzip it, there are no corrupted filenames as described in [LPS-148397](https://issues.liferay.com/browse/LPS-148397)

Jira Ticket: https://issues.liferay.com/browse/LPS-157175

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: portal-web/test/functional/com/liferay/portalweb/macros/DMFolder.macro [hunk 1]
```diff
@@ -251,6 +251,18 @@ definition {
 		AssertElementNotPresent(locator1 = "DocumentsAndMedia#DESCRIPTIVE_LIST_FOLDER_ICON");
 	}
 
+	macro downloadViaEllipsisMenu {
+		ProductMenu.gotoPortlet(
+			category = "Content & Data",
+			portlet = "Documents and Media");
```
Comment (austinchiang) [inline portal-web/test/functional/com/liferay/portalweb/macros/DMFolder.macro:257]: This isn't needed in the macro. If you need to refresh the UI before clicking on the folder ellipsis you can use the Refresh function at the test case level to do it

File: portal-web/test/functional/com/liferay/portalweb/macros/DMFolder.macro [hunk 2]
```diff
@@ -251,6 +251,18 @@ definition {
...
+
+		Click(
+			key_dmFolderName = "${dmFolderName}",
+			locator1 = "DocumentsAndMedia#FOLDER_ELLIPSIS");
```
Comment (austinchiang) [inline portal-web/test/functional/com/liferay/portalweb/macros/DMFolder.macro:261]: Can you reuse Icon#ROW_VERTICAL_ELLIPSIS here? you would just need to declare the folder name

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/documentmanagement/documentsadministration/DMFileEntry.testcase [hunk 3]
```diff
@@ -2893,6 +2893,54 @@ definition {
...
+			dmDocumentFile = "Document_1.doc",
+			dmDocumentTitle = "Document_1.doc");
+
+		DMFolder.addCP(dmFolderName = "日本語フォルダ2");
+
+		DMNavigator.gotoFolder(dmFolderName = "日本語フォルダ2");
+
+		DMDocument.addCP(
+			dmDocumentFile = "Document_2.doc",
+			dmDocumentTitle = "Document_2.doc");
```
Comment (austinchiang) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/documentmanagement/documentsadministration/DMFileEntry.testcase:2915]: Please use existing JSON macros for these

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/documentmanagement/documentsadministration/DMFileEntry.testcase [hunk 4]
```diff
@@ -2893,6 +2893,54 @@ definition {
...
+		var tempDir = PropsUtil.get("output.dir.name");
+
+		var zipFile = FileUtil.exists("${tempDir}/日本語フォルダ1.zip");
+		var document1 = FileUtil.exists("${tempDir}/Document_1.doc");
+		var folder2 = FileUtil.exists("${tempDir}/日本語フォルダ2");
+		var document2 = FileUtil.exists("${tempDir}/日本語フォルダ2/Document_2.doc");
+
+		if (("${zipFile}" == "false") || ("${document1}" == "false") || ("${folder2}" == "false") || ("${document2}" == "false")) {
+			fail("There was an error on DMFileEntry.NoCorruptedNameInDownloadedZip test, filenames are not the expected.");
+		}
```
Comment (austinchiang) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/documentmanagement/documentsadministration/DMFileEntry.testcase:2932]: Please move into a macro and make it reusable

**Review by austinchiang — CHANGES_REQUESTED** [review]

**Pull request closed without merging**